department-of-veterans-affairs/vets-website
# Check in/89875/add start end date to request

**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?

If the folder you changed contains a `manifest.json`, search for its `entryName` in the content-build [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) (the `entryName` there will match).

If an entry for this folder exists in content-build and you are:
1. **Deleting a folder**: Delete the application entry in [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) and merge that PR **before** this one
- _Add the link to your merged content-build PR here_

2. **Renaming or moving a folder**: Update the entry in the [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json), but do not merge it until your vets-website changes here are merged. The content-build PR must be merged immediately after your vets-website change is merged in to avoid CI errors with content-build (and Tugboat).

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?

## Summary

- Adds start and end date parameters to the upcoming appointments call
- Modifies mocks and response handling to align with recent response changes from that call

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
[department-of-veterans-affairs/va.gov-team#89875](https://github.com/department-of-veterans-affairs/va.gov-team/issues/89875)

## Testing done

- API was called with no paramters
- Enable the mock feature flag for upcoming appointments in src/applications/check-in/api/local-mock-api/mocks/v2/feature-toggles/index.js
- visit the appointments page of the app
- Upcoming appointments still display with all the necessary data

## Screenshots
![Screenshot 2024-08-08 at 13-13-56 check-in Veterans Affairs](https://github.com/user-attachments/assets/c80aa3dd-87b2-407d-b40a-214a0ba141a8)
![Screenshot 2024-08-08 at 13-13-44 check-in Veterans Affairs](https://github.com/user-attachments/assets/6c67728f-faf2-44b8-a36a-8efed9287ff0)
![Screenshot 2024-08-08 at 1 13 11 PM](https://github.com/user-attachments/assets/3f84f4f4-a585-4890-8caf-0e727e1b87e5)



## What areas of the site does it impact?

Day of and pre check in 

## Acceptance criteria

- [ ] Sends the correct request
- [ ] Displays the correct data

### Quality Assurance & Testing


### Error Handling


### Authentication


## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_
